### PR TITLE
Added voting to the governance SCORE

### PR DESCRIPTION
**Pull request by dbrehmer**
A first draft for voting functionality with only the vote to enable dividends distribution possible.

### CODE REVIEW

File: core_contracts/governance/governance.py [hunk 1]
```diff
@@ -28,30 +28,127 @@ class Governance(IconScoreBase):
...
+        self._vote_actions = DictDB('vote_actions', db, str)
+        self._vote_count = VarDB('vote_count', db, int)
 
     def on_install(self) -> None:
         super().on_install()
         self._launched.set(False)
 
     def on_update(self) -> None:
         super().on_update()
+        self._minimum_vote_duration.set(42300)
```
Comment (ajaya-man) [inline core_contracts/governance/governance.py:47]: I think the number is trying to resemble a day which is actually 43200.

File: core_contracts/governance/governance.py [hunk 2]
```diff
@@ -28,30 +28,108 @@ class Governance(IconScoreBase):
...
+        self._votes[vote_index][0]['majority'] = majority
+        self._votes[vote_index][0]['last_block'] = self.block.height + duration
+        self._vote_actions[vote_index] = actions
+
+    @external
+    def castVote(self, name: str, vote: bool) -> None:
+        """
+        Casts a vote in the named poll.
+        """
+        if name not in self._votes[0] or not self._votes[self._votes[0][name]['id']][0]['active']:
```
Comment (ajaya-man) [inline core_contracts/governance/governance.py:91]: self._votes is a dictdb. Can't perform `in` check here.

File: core_contracts/governance/governance.py (hunk 1 above)
Comment (dbrehmer) [reply core_contracts/governance/governance.py]: Changed to be in day units (1).

File: core_contracts/governance/governance.py (hunk 2 above)
Comment (dbrehmer) [reply core_contracts/governance/governance.py]: DictDB has a __contains__ method, so `in` works.

**Comment by dbrehmer** [comment]
Deployed and tested on testnet with vote creation. checkVote gives an error though due to the revert in the stakedBalanceOfAt method on the balance token SCORE. Thinking that we should avoid reverts in readonly methods and just return a zero in this case.

File: core_contracts/governance/governance.py [hunk 3]
```diff
@@ -72,7 +72,7 @@ def cancelVote(self, name: str) -> None:
         Cancels a vote, in case a mistake was made in its definition.
         """
         vote_index = ProposalDB.proposal_id(name, self.db)
-        if vote_index > 0:
+        if vote_index == 0:
```
Comment (ajaya-man) [inline core_contracts/governance/governance.py:75]: I think this check is done in other parts also, I will correct on those.

**Review by dbrehmer — COMMENTED** [review]
Latest changes look good.

**Pull request merged**